### ISSUE TITLE
Hardlinks are not created by default

### ISSUE THREAD

**Issue by kayahr** [pnpm/pnpm#2355]
### pnpm version: 4.9.3

`pnpm install` doesn't create hardlinks at all by default but it works when I use `pnpm install --package-import-method=hardlink`.

According to the documentation the default import method is **auto** which should do hardlinking when possible and fall back to copying if hardlinking fails. Hardlinking obviously works on my system when selected manually but still the **auto** setting falls back to copying the files here. No warnings are displayed telling me why pnpm falls back to copying by default. That looks wrong.

Repro with default settings:

```sh
$ mkdir /tmp/test1
$ cd /tmp/test1
$ pnpm i lodash
Packages: +1
+
Resolving: total 1, reused 1, downloaded 0, done

dependencies:
+ lodash 4.17.15
$ ls -i node_modules/lodash/package.json  ~/.pnpm-store/2/registry.npmjs.org/lodash/4.17.15/node_modules/lodash/package.json
25921547 ~/.pnpm-store/2/registry.npmjs.org/lodash/4.17.15/node_modules/lodash/package.json
25944234 node_modules/lodash/package.json
```

Notice that the inodes are different, so no hardlinking was done, files were copied.

Now try again with `--package-import-method=hardlink` parameter:

```sh
$ mkdir /tmp/test2
$ cd /tmp/test2
$ pnpm i lodash --package-import-method=hardlink
Packages: +1
+
Resolving: total 1, reused 1, downloaded 0, done

dependencies:
+ lodash 4.17.15
$ ls -i node_modules/lodash/package.json  ~/.pnpm-store/2/registry.npmjs.org/lodash/4.17.15/node_modules/lodash/package.json
25921547 ~/.pnpm-store/2/registry.npmjs.org/lodash/4.17.15/node_modules/lodash/package.json
25921547 node_modules/lodash/package.json
```

Now inodes are the same, hardlinking worked.

I would expect that the default **auto** setting creates hardlinks or at least prints a warning why it falls back to copying files.

 - `node -v` prints: v13.8.0
 - Debian Linux 10 using a single btrfs file system.

**Comment by kayahr** [comment]
I checked the source code and discovered that pnpm uses copy-on-write clones if filesystem supports it (and btrfs does) and falls back to hardlinking if it doesn't. So cloning is used here instead of hardlinking. I guess that's fine, so there is no bug here and I'm closing the issue. Sorry for interruption.

I still prefer hardlinking because this can easily be detected and shows the correct disk usage when using `du` or `ncdu`. So I added `package-import-method=hardlink` to my `.npmrc` and I'm happy now :)

**Comment by zkochan** [comment]
Probably we need to mention this in the docs. In docs we mainly mention only hard links because copy-on-write is not widely supported

**Comment by jhalag** [comment]
Just posting to say I just ran into this. Using BTRFS, and was trying out pnpm for the first time, trying to figure out why no hard links were being created. Relieved to see it's using reflinks, but the documentation did not seem to immediately reveal that fact (though digging in I did see a reflink option in the configuration, which made me suspect).

As CoW is becoming more popular (BTRFS is standard on Fedora now as of 33 I believe) - I believe it would be prudent to add that to the documentation. Perhaps just an extra line to the FAQ section that describes hardlinks?

**Comment by zkochan** [comment]
It is documented here: https://pnpm.js.org/en/npmrc#package-import-method

but maybe it is confusing that the [motivation](https://pnpm.js.org/en/motivation) page only mentions hard links

**Comment by jhalag** [comment]
Yes, the package-import-method config is where (once I started digging) I saw a hint that it might be reflinking as opposed to hardlinking.

In my troubleshooting, I first was reading over pages such as the motivation page you linked, and the [faq](https://pnpm.js.org/en/faq) which describes the hardlinking process, it does not mention reflinks. So when I went looking around (with ls -l) I thought it was broken at first.

Even just changing that second sentence in the FAQ to [...] Hard links _(or reflinks, if supported by the filesystem)_ point to the same place on the disk where the original files are. [...] might be helpful?
